### PR TITLE
initial archive access code

### PR DESCRIPTION
**Pull request by ehneilsen**
Adding tools for getting run and family information and downloading opsim runs and summary metrics.

### CODE REVIEW

**Review by rhiannonlynne — APPROVED** [review]
Just documentation updates (and a rebase on main).

File: rubin_sim/maf/runComparison/archive.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+        If it is set to `None`, the data is loaded from the URL specified
+        by the `archive.RUNS_SOURCE` constant.
+
+    Returns
+    -------
+    runs : `pandas.DataFrame`
+        ``run``
+            The index of the DataFrame is the project-standard name for the run
+            (`str`)
+        ``familie``
```
[the remark this answers is not in the document]
Comment (ehneilsen) [reply rubin_sim/maf/runComparison/archive.py]: fixed.

File: rubin_sim/maf/runComparison/archive.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+        by the `archive.METRIC_SET_SOURCE` constant.
+
+    Returns
+    -------
+    metric_sets : `pandas.DataFrame`
+        ``metric_set``
+            The 1st level of the index is the name of a set of metrics (`str`).
+        ``metric``
+            The 2rd level of the index is the full name of the metric (`str`).
+        ``metric``
```
[the remark this answers is not in the document]
Comment (ehneilsen) [reply rubin_sim/maf/runComparison/archive.py:244]: The first lists "metric" as a level of the index, the second as a column in the DataFrame. Yes, this is redundant and uses more memory than strictly necessary, but can make the user's life easier: some operations are more convenient (and sometimes faster) if the data is in an index level, others are more convenient if it is in a column. In this case, I found myself wanting to use "metric" both ways, so I just put it both places.

File: rubin_sim/maf/runComparison/archive.py [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+                warnings.warn(f'Metric "{metric}" not in summary, skipping')
+                metrics.remove(metric)
+
+        if metric_order == "summary":
+            metrics = [m for m in all_summaries.columns if m in metrics]
+
+    summaries = all_summaries.loc[runs, metrics]
+    summaries.columns.name = "metric"
+    summaries.index.name = "run"
+    return summaries
```
[the remark this answers is not in the document]
Comment (ehneilsen) [reply rubin_sim/maf/runComparison/archive.py:390]: Okay, done.

**Pull request merged**